src/apps/loan-list/utils/pagination.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState, FC } from "react";
+import { LoanV2 } from "../../../core/fbs/model/loanV2";
+import { getStackedSearchItems, getSearchItems } from "./helpers";
+import ResultPager from "../../../components/result-pager/result-pager";
+import { pageSize } from "../../../core/configuration/pagesize.json";
+import LoanListItems from "../list/loan-list-items";
+import { GetMaterialManifestationQuery } from "../../../core/dbc-gateway/generated/graphql";
+import { LoanDetailsV2 } from "../../../core/fbs/model";
+
+interface PaginationProps {
+  selectModalMaterial: ({
+    material,
+    loanDetails
+  }: {
+    material: GetMaterialManifestationQuery | undefined | null;
+    loanDetails: LoanDetailsV2;
+  }) => void;
+  openModalDueDate: (dueDate: string) => void;
+  view: string;
+  hitcount: number;
+  dueDates: string[];
+  loans: LoanV2[];
+}
+
+const Pagination: FC<PaginationProps> = ({
+  view,
+  hitcount,
+  dueDates,
+  loans,
+  openModalDueDate,
+  selectModalMaterial
+}) => {
+  const [searchItemsShown, setSearchItemsShown] = useState(pageSize);
+  const [displayedLoans, setDisplayedLoans] = useState<LoanV2[]>();
+  const [page, setPage] = useState<number>(0);
+
+  const setPageHandler = () => {
+    if (loans) {
+      const currentPage = page + 1;
+      const itemsOnPage = (currentPage + 1) * pageSize;
+      setPage(currentPage);
+      setSearchItemsShown(itemsOnPage);
+    }
+  };
```

Comment:
Isn't it possible to make this part generic as well? I am just freestyling here, but could it be possible to create a custom hook that returns the pager component with a contract like this:
```
const [itemsShown, hitcount, pagehandler, ResultPager] = usePager(pageSize);
```
and later on in the rendering:
```
return (
  <div>
    <AlistOfSomething />
    <ResultPager itemsShown={itemsShown} hitcount={hitcount} pagehandler={pagehandler} />
  </div>
);

Reply:
I am not entirely sure what you mean, create a hook that returns the pager component _instead of_ pagination.tsx? And to cover the search results as well?